server/src/main/java/org/solarbank/server/EnergySavingController.java

```diff
@@ -14,22 +17,27 @@
 @RestController
 @RequestMapping("/v1.0/api")
 public class EnergySavingController {
-
     private final CalculateService calculateService;
+    private final NasaClient nasaClient;
 
     @Autowired
-    public EnergySavingController(CalculateService calculateService) {
+    public EnergySavingController(CalculateService calculateService, NasaClient nasaClient) {
         this.calculateService = calculateService;
+        this.nasaClient = nasaClient;
     }
 
     @PostMapping("/calculate")
     public ResponseEntity<CalculateResult> userInput(
         @Valid @RequestBody CalculateRequest calculateRequest
     ) {
+
+        Map<String, Double> nasaData = nasaClient.getNasaData(calculateRequest.getLocation());
```

Comment:
This will make the call to the Nasa API and return nasaData that is used to do the energy saving calculations. I was unsure if I should call the function here or from processCalculateRequest() method below.